File: 260_Single_Number_III/singleNumber.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    vector<int> singleNumber(vector<int>& nums) {
        int len = nums.size();
        int num0 = nums[0];
        int num1 = nums[1];
        if(len <= 2) {
            return nums;
        }
        for(int i=0;i<len;i++) {
            if(num0 == nums[i] && i!=0) {
                num0 = nums[i++];
            }
        }
        for(int i=0;i<len;i++) {
            if((num1 == nums[i] || num1 != num0) && i!=1) {
                num1 = nums[i++];
            }
        }
        vector<int> res;
        res.push_back(num0);
        res.push_back(num1);
        return res;
    }
};

int main() {
    Solution solution;
    int nums[] = {0,0,1,2};
    vector<int> numsVec(&nums[0], &nums[3]);
    vector<int> resVec = solution.singleNumber(numsVec);
    cout << numsVec << endl;
    return 0;
}

File: 260_Single_Number_III/singleNumber.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var singleNumber = function(nums) {
    var len = nums.length;
    if(len <= 2) {
        return nums;
    }
    var flags = {};
    var i=0;
    for(i=0;i<len;i++) {
        var curNum = nums[i];
        var curCount = flags[curNum] || 0;
        curCount++;
        flags[curNum] = curCount;
    }
    var res = [];
    for(var k in flags) {
        if(flags[k] === 1) {
            res.push(parseInt(k, 10));
        }
        if(res.length === 2) {
            return res;
        }
    }
};

var nums = [1,2,1,3,2,5];
console.log(singleNumber(nums));